### ISSUE TITLE
Implement autofill search UI

### ISSUE THREAD

**Issue by sandysage** [mozilla-lockbox/lockbox-android#467]
### Acceptance Criteria

The user should be able to search for the desired login entry if the correct one is not presented -- including none.

### Visual Design

**InVision**
App (no matches) - https://mozilla.invisionapp.com/share/3SN35UHWJDX#/screens/317568427
App (single match - https://mozilla.invisionapp.com/share/3SN35UHWJDX#/screens/317564412
App (multiple matches) - https://mozilla.invisionapp.com/share/3SN35UHWJDX#/screens/317564409
Search (initial) - https://mozilla.invisionapp.com/share/3SN35UHWJDX#/screens/317564411
Search (single match) - https://mozilla.invisionapp.com/share/3SN35UHWJDX#/screens/317564410
Search (multiple matches) - https://mozilla.invisionapp.com/share/3SN35UHWJDX#/screens/319298324
Search (no matches) - https://mozilla.invisionapp.com/share/3SN35UHWJDX#/350129873_Autofill_-_App_-_Search_-_Text_Entered_-_No_Results

Depends on #421

**Comment by sashei** [comment]
Copied from #421 comment by @linuxwolf : 
[ cc @changecourse for visibility ]

> * What should the search dialog look like when there are a lot of entries matching what's typed in the search box (e.g., the user only types `t`)? How should it scroll, and how high should it get?

Briefly discussed in standup that it should scroll, but not these particulars.

@changecourse will have better thoughts, though to start with some strawpeople:

* I think it would be desirable to keep the search bar visible at all times, and scrolling limited to just the login entries
* while it would be nice to keep all of the dialog visible even when the virtual keyboard is open, I don't know how feasible that is.  If not feasible, I would suggest the bottom limit be the same distance as the side margins?

> * What should the behavior be when tapping outside of the search box?

Briefly discussed in standup that it should behave like other dialogs, in that clicking outside should dismiss it

**Comment by sashei** [comment]
There's another question here around the "no entries" state -- do we implement "No entries found"? What if the user doesn't have any entries saved into Lockbox at all?

**Comment by sashei** [comment]
Is there anything we want to display when the user hasn't logged into lockbox before? Maybe there's an onboarding-in-autofill story that I can't find, just wanted to make sure that this case was captured / thought about.

**Comment by changecourse** [comment]
> What should the search dialog look like when there are a lot of entries matching what's typed in the search box (e.g., the user only types t)? How should it scroll, and how high should it get?

Unless there are issues around performance, I would want this to function where it begins to populate a list as soon as a single character is selected, e.g., I select 't' and see the 20+ entries in a list that begin with a T, in alphabetical order. Even if there is a delay associated with populating such a large, scrollable list, we can assume that someone will type the first few letters quickly, so that in most instances, the list will be parsed down to only a handful of results, e.g., 'twi' showing 2-3 potentially. 


> I think it would be desirable to keep the search bar visible at all times, and scrolling limited to just the login entries

Agreed 💯 


> What should the behavior be when tapping outside of the search box?

Agree with Matt, "it should behave like other dialogs, in that clicking outside should dismiss it"


> What if the user doesn't have any entries saved into Lockbox at all?

Good question! I think the functionality should remain the same, but that we handle two different use cases in the same manner, matching entries and 0 total entries, the same. I've included a screenshot here, and will add to InVision links in description above. We could look to handling the 0 total entries in a more elegant way in the future, but for now, I think this will suffice. 

<img width="434" alt="screen shot 2019-03-04 at 9 46 28 am" src="https://user-images.githubusercontent.com/6164801/53748430-8e804980-3e62-11e9-8fd1-8060b4a7f169.png">

Please let me know if there's anything else I can help with regarding this effort, @sashei :)

**Comment by changecourse** [comment]
>Is there anything we want to display when the user hasn't logged into lockbox before? Maybe there's an onboarding-in-autofill story that I can't find, just wanted to make sure that this case was captured / thought about. 

My assumption here would be that if Lockbox hasn't been launched ever, it would be highly unlikely that Lockbox would be set as an autofill provider, so they wouldn't see any dialogs for us associated with autofill. @sandysage we could create an issue to track this edge case if you agree, for the future.

**Comment by sandysage** [comment]
> My assumption here would be that if Lockbox hasn't been launched ever, it would be highly unlikely that Lockbox would be set as an autofill provider, so they wouldn't see any dialogs for us associated with autofill. @sandysage we could create an issue to track this edge case if you agree, for the future.

Oh! Good catch y'all. Yes, I'll create that separate story now.
